swift/apple/FirezoneKit/Sources/FirezoneKit/Models/Favorites.swift

```diff
@@ -1,45 +1,51 @@
 import Foundation
 
 public class Favorites: ObservableObject {
-  private static let key = "favoriteResourceIDs"
+  private let key = "favoriteResourceIDs"
   @Published private(set) var ids: Set<String>
 
-  public init() {
-    ids = Favorites.load()
+  public init () {
+    self.ids = Set()
+    Task { await self.ids = load() }
   }
 
   func contains(_ id: String) -> Bool {
     return ids.contains(id)
   }
 
-  func reset() {
-    objectWillChange.send()
-    ids = Set()
-    save()
+  func load() async -> Set<String> {
+    let ids = await withCheckedContinuation { continuation in
+      continuation.resume(returning: UserDefaults.standard.stringArray(forKey: key))
+    }
+
+    if let ids {
+      return Set(ids)
+    }
+
+    return Set()
   }
 
-  func add(_ id: String) {
-    objectWillChange.send()
-    ids.insert(id)
-    save()
+  func reset() async {
+    self.ids = Set()
+    await save()
   }
 
-  func remove(_ id: String) {
-    objectWillChange.send()
-    ids.remove(id)
-    save()
+  func add(_ id: String) async {
+    self.ids.insert(id)
+    await save()
   }
 
-  private func save() {
-    // It's a run-time exception if we pass the `Set` directly here
-    let ids = Array(ids)
-    UserDefaults.standard.set(ids, forKey: Favorites.key)
+  func remove(_ id: String) async {
```

Comment:
Is making these `async` the right direction? It seems like saving them to disk is a side-effect of updating our in-memory model. Perhaps we should spawn a task here already?

Reply:
Yeah, that works too, will update.